lib/luadata.c

```diff
@@ -244,6 +248,42 @@ static int luadata_setstring(lua_State *L)
 	return 0;
 }
 
+/***
+* Resizes the memory block represented by the data object.
+* If the object is a network packet (SKB), it uses skb_put() to expand 
+* or skb_trim() to shrink the buffer. For raw buffers, it updates the size.
+* @function resize
+* @tparam integer new_size The desired size of the memory block in bytes.
+* @raise Error if the data object is read-only or if resize fails.
+*/
+static int luadata_resize(lua_State *L)
+{
+    luadata_t *data = luadata_check(L, 1);
+    size_t new_size = (size_t)luaL_checkinteger(L, 2);
+    
+    luadata_checkwritable(L, data);
+
+    if (data->opt & LUADATA_OPT_SKB) {
+		struct sk_buff *skb = LUADATA_TOSKB(data);
+		if (new_size > data->size) {
+			size_t needed = new_size - data->size;
+			if (skb_tailroom(skb) < needed)
+				luaL_error(L, "insufficient tailroom for resize");
+			skb_put(skb, needed);
+		}
+		else if (new_size < data->size)
+			skb_trim(skb, new_size);
```

Comment:
wondering if it would be better to move this block to an aux func. What do you think? feel free to choose =). No strong feelings on this..